pkg/apis/vsphere/v1/template_helpers.go

```diff
@@ -0,0 +1,73 @@
+package v1
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+
+	"go.anx.io/go-anxcloud/pkg/api"
+	"go.anx.io/go-anxcloud/pkg/api/types"
+	corev1 "go.anx.io/go-anxcloud/pkg/apis/core/v1"
+)
+
+var (
+	// ErrTemplateNotFound is returned when the named template was not found at a given location
+	ErrTemplateNotFound = fmt.Errorf("%w: named template was not found at specified location", api.ErrNotFound)
+)
+
+const (
+	// LatestTemplateBuild is used to find the template with the highest build number
+	LatestTemplateBuild = "latest"
+)
+
+// FindNamedTemplate retrieves a template by name and build at a specified location.
+// Empty and LatestTemplateBuild build identifier will yield the highest available build.
+// It returns ErrTemplateNotFound if no matching template was found.
+func FindNamedTemplate(ctx context.Context, a api.API, name, build string, location corev1.Location) (*Template, error) {
+	var (
+		match   *Template
+		buildNo = -1
+		latest  = build == "" || build == LatestTemplateBuild
+	)
+
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	var channel types.ObjectChannel
+
+	err := a.List(ctx, &Template{Type: TypeTemplate, Location: location}, api.ObjectChannel(&channel))
+	if err != nil {
+		return nil, err
+	}
+
+	for res := range channel {
+		var template Template
+		err := res(&template)
+		if err != nil {
+			return nil, err
+		}
+
+		if template.Name != name {
+			continue
+		}
+
+		if latest {
+			currentTemplateBuildNo, _ := strconv.Atoi(template.Build[1:])
```

Comment:
please check if `strconv.Atoi` was successful

Reply:
Maybe give `Template` a method `BuildNumber() int`, you could use this here and users of go-anxcloud could easily reuse the same parsing logic for something different.

Reply:
Should the `BuildNumber` method also return an error when Atoi wasn't successful or default to zero-val?

Reply:
hmm... it should either try to log a warning (`logr.FromContextOrDiscard` or something) and skip the template or error out - defaulting to zero-val isn't good

Either way, it should only do that when "latest" template is requested.

Let's return an error